contracts/Elections.sol

```diff
@@ -111,13 +113,33 @@ contract Elections is IElections, ContractRegistryAccessor {
 		getComplianceCommitteeContract().memberReadyToSync(sender, false);
 	}
 
-	function notifyDelegationChange(address newDelegatee, address prevDelegatee, uint256 newStakePrevDelegatee, uint256 newStakeNewDelegatee, uint256 prevGovStakePrevDelegatee, uint256 prevGovStakeNewDelegatee) onlyDelegationsContract external {
+	function notifyDelegationChange(address delegator, uint256 delegatorSelfStake, address newDelegate, address prevDelegate, uint256 prevDelegateNewTotalStake, uint256 newDelegateNewTotalStake, uint256 prevDelegatePrevTotalStake, bool prevSelfDelegatingPrevDelegate, uint256 newDelegatePrevTotalStake, bool prevSelfDelegatingNewDelegate) onlyDelegationsContract external {
+		require(newDelegate != prevDelegate, "in a delegation change the delegate must change");
 
-        _applyDelegatedStake(prevDelegatee, newStakePrevDelegatee);
-		_applyDelegatedStake(newDelegatee, newStakeNewDelegatee);
+		if (delegator == newDelegate) { // delegator != prevDelegate
```

Comment:
This if block takes care of adding/subtracting the self stake of the delegator from the `totalGovernanceStake`, but what about the delegated stake? If a delegator became self delegating, it's delegated stake should also be added. Similarly when becoming not self delegated. Did I miss anything?

Reply:
correct